[4.0] Modify display and tip for Edit Article in Frontend

infograf768 opened the pull request:
### Summary of Changes
Since Email to a friend and Print article have been taken off from J4, we do not need the anymore the Cog and dropdown to choose among actions. We now need only the Edit Article when authorised to do so.
Use of `role="tooltip"]` instead of bootstrap tips.
Cassiopea `[role="tooltip"]` is modified for the look. Placement of tip specific to the article edit id.

Should also be OK for a11y. TO CHECK please as I'm not a specialist.

### Testing Instructions
3 situations:
1. Edit Article when article published // simplified tip
2. Edit Article when article is unpublished // simplified tip
3. Article is checked out by another user

In each case, hover the icon to see the tip.

patch, run NPM


### Actual result BEFORE applying this Pull Request
<img width="1221" alt="Screen Shot 2020-09-07 at 18 35 48" src="https://user-images.githubusercontent.com/869724/92407005-bcbaa580-f139-11ea-8aa7-d5cb92e926c5.png">

<img width="1208" alt="Screen Shot 2020-09-07 at 18 37 57" src="https://user-images.githubusercontent.com/869724/92407046-d8be4700-f139-11ea-9823-f10e47ae7536.png">

<img width="1213" alt="Screen Shot 2020-09-07 at 18 37 23" src="https://user-images.githubusercontent.com/869724/92407066-e1168200-f139-11ea-8b98-22f3143e191b.png">

### Expected result AFTER applying this Pull Request
<img width="866" alt="Screen Shot 2020-09-07 at 18 15 40" src="https://user-images.githubusercontent.com/869724/92407237-4a969080-f13a-11ea-92f1-0c4b989feb9a.png">

<img width="883" alt="Screen Shot 2020-09-07 at 18 14 48" src="https://user-images.githubusercontent.com/869724/92407284-669a3200-f13a-11ea-9f8a-9a2bf6b688fc.png">

<img width="905" alt="Screen Shot 2020-09-07 at 18 13 31" src="https://user-images.githubusercontent.com/869724/92407298-744fb780-f13a-11ea-82d9-bf7954eb213c.png">





### Note
If this is OK, will do for Contact

infograf768 committed:
[4.0] Modify display and tip for EDit Article in Frontend

richard67 commented on administrator/components/com_content/src/Service/HTML/Icon.php:
```diff
@@ -145,58 +144,23 @@ public function edit($article, $params, $attribs = array(), $legacy = false)
+		$text = LayoutHelper::render('joomla.content.icons.edit', array('article' => $article, 'tooltip' => $tooltip, 'legacy' => $legacy));
 
-		$attribs['title']   = Text::_('JGLOBAL_EDIT_TITLE');
+		//$attribs['title']   = Text::_('JGLOBAL_EDIT_TITLE');
```
Shall this commented out line be removed?

infograf768 replied:
yep

infograf768 committed:
take off commented line

brianteeman commented:
I have tested this item :red_circle: unsuccessfully on 741f9ca6be159bd9250ce3191be3710d8d09ec54<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30588">issues.joomla.org/tracker/joomla-cms/30588</a>.</sub>

brianteeman commented:
The accessibility changes you made are completely wrong
1. never add role=button to a link
2. incorrect usage of aria-live (and not needed)

infograf768 commented:
> The accessibility changes you made are completely wrong
 >   never add role=button to a link
 >   incorrect usage of aria-live (and not needed)

It was useless to set unsuccessful as I noted in the description `TO CHECK please as I'm not a specialist.`

Getting rid of  these 2 aspects. Is there anything else?

infograf768 committed:
correcting wrong a11y parts

brianteeman commented:
I cannot replicate your screenshots - looks like you might be missing a class on the "tooltip" div

![image](https://user-images.githubusercontent.com/1296369/92447779-9f282300-f1af-11ea-936b-7a42774d3d25.png)

![image](https://user-images.githubusercontent.com/1296369/92447840-b109c600-f1af-11ea-9cbe-50e8cf6cf5eb.png)

brianteeman commented:
Also because the link is called edit and the tooltip begins with edit it is announced by a screenreader as
Edit Edit ....

infograf768 commented:
NPM ci is necessary.

Class is not missing, role tooltip is a class de facto
```
// set up hidden tooltip
[role="tooltip"]:not(.show) {
  z-index: $zindex-tooltip;
  display: none;
  max-width: 100%;
  padding: .5em;
  margin: .5em;
  color: $black;
  background: $white-offset;
  border: 1px solid $gray-600;
  border-radius: $border-radius;
  box-shadow: 0 0 .5rem rgba(0,0,0,.8);

  &[id^=editarticle-] {
    margin-inline-start: -10em;
  }

  [dir=ltr] & {
    text-align: left;
  }

  [dir=rtl] & {
    text-align: right;
  }
}


but it looks like I can't apply anymore this .diff in eclipse for the 3 layouts files
Checking that aspect now here.

Can you explain 
> Also because the link is called edit and the tooltip begins with edit it is announced by a screenreader as

Not sure what you mean by `the link is called edit`. Can you explain? And what it should be?

infograf768 commented:
I see now why this PR does ot apply anymore here..
https://github.com/joomla/joomla-cms/pull/30543 was merged

infograf768 commented:
The git apply issue solved itself here.. :)

infograf768 commented:
> Also because the link is called edit and the tooltip begins with edit it is announced by a screenreader as Edit Edit ....

Got it now. I don't think it is a big deal.

brianteeman commented:
That would be a decision by the accessibility team

infograf768 commented:
Stefan made suggestions. Will modify stuff shortly.

infograf768 committed:
various improvements

infograf768 commented:
Modified with various improvements. aria-hidden true. Simplified tooltips.
Now we get:

<img width="933" alt="Screen Shot 2020-09-08 at 17 28 10" src="https://user-images.githubusercontent.com/869724/92496967-1afa8d80-f1f9-11ea-82ae-f1cf27a77519.png">

<img width="866" alt="Screen Shot 2020-09-08 at 17 24 42" src="https://user-images.githubusercontent.com/869724/92497037-2a79d680-f1f9-11ea-840e-cc5722591293.png">

<img width="880" alt="Screen Shot 2020-09-08 at 17 26 56" src="https://user-images.githubusercontent.com/869724/92497106-39608900-f1f9-11ea-8521-59e223dc883f.png">

hans2103 commented:
I have tested this item :white_check_mark: successfully on f9fd8ed3c70e5736ff3caeb357275f62466b9b05<br /><br />I see the simplified tip in all three situations<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30588">issues.joomla.org/tracker/joomla-cms/30588</a>.</sub>

brianteeman commented on layouts/joomla/content/icons/edit_lock.php:
```diff
@@ -9,11 +9,15 @@
 ?>
-<span class="hasTooltip fas fa-lock fa-fw" title="<?php echo HTMLHelper::tooltipText($tooltip . '', 0); ?>"></span>
-<?php echo Text::_('JLIB_HTML_CHECKED_OUT'); ?>
+<span class="hasTooltip fas fa-lock"  aria-hidden="true"></span>
```
```suggestion
<span class="hasTooltip fas fa-lock" aria-hidden="true"></span>
```

remove space

infograf768 replied:
done

brianteeman commented on templates/cassiopeia/scss/blocks/_form.scss:
```diff
@@ -74,10 +72,17 @@ td .form-control {
+  background: $white-offset;
+  border: 1px solid $gray-600;
+  border-radius: $border-radius;
+  box-shadow: 0 0 .5rem rgba(0,0,0,.8);
```
```suggestion
  box-shadow: 0 0 .5rem rgba(0, 0, 0, .8);
```

infograf768 replied:
done

brianteeman commented on layouts/joomla/content/icons/edit.php:
```diff
@@ -25,7 +24,11 @@
-<span class="hasTooltip fas fa-<?php echo $icon; ?> fa-fw" title="<?php echo HTMLHelper::tooltipText(Text::_('COM_CONTENT_EDIT_ITEM'), $overlib, 0, 0); ?>"></span>
-<?php echo Text::_('JGLOBAL_EDIT'); ?>
+<span class="fas fa-<?php echo $icon; ?>" aria-hidden="true"></span>
+ <?php echo Text::_('JGLOBAL_EDIT'); ?>
```
```suggestion
	<?php echo Text::_('JGLOBAL_EDIT'); ?>
```

space to tab

infograf768 replied:
done

brianteeman commented:
Don't know why but the role=tooltip css is present in my unminified css file but NOR in the minified one

chmst commented:
@brianteeman did you use the patchtester? I have the same effect, but think that this not related to the PR.

brianteeman commented:
@chmst I did but I cant see how that would make a difference. If the code is present in the non-minified css then the patch has worked

brianteeman commented:
found the problem - its not related to patchtester - creating a seperate issue for it

brianteeman commented:
I have tested this item :white_check_mark: successfully on f9fd8ed3c70e5736ff3caeb357275f62466b9b05<br /><br />Marking as successful as the minification issue is unrelated to the actual pr<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30588">issues.joomla.org/tracker/joomla-cms/30588</a>.</sub>

infograf768 committed:
cs

infograf768 committed:
cs

infograf768 commented:
> Don't know why but the role=tooltip css is present in my unminified css file but NOR in the minified one

<img width="629" alt="Screen Shot 2020-09-09 at 07 39 21" src="https://user-images.githubusercontent.com/869724/92558817-a6af0100-f26f-11ea-8bef-a60cdecd679f.png">

infograf768 commented:
RTC as last changes were small cs corrections.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30588">issues.joomla.org/tracker/joomla-cms/30588</a>.</sub>

laoneo commented:
Thanks!

@wilsonge do we need to document the change that the helper function got removed in the upgrade guide?

infograf768 commented:
Will now make similar changes for the Edit Contact.

wilsonge commented:
@laoneo I guess it can't hurt to document it although I suspect it's a touch overkill
